flocker/control/_persistence.py

```diff
@@ -158,20 +295,50 @@ def __init__(self, reactor, path):
         MultiService.__init__(self)
         self._path = path
         self._change_callbacks = []
+        self._config_class = getattr(
+            sys.modules[__name__],
+            u"Configuration_V" + unicode(_CONFIG_VERSION)
+        )
         LeaseService(reactor, self).setServiceParent(self)
 
     def startService(self):
         if not self._path.exists():
             self._path.makedirs()
-        self._config_path = self._path.child(b"current_configuration.v1.json")
+        self.load_configuration()
+        MultiService.startService(self)
+        _LOG_STARTUP(configuration=self.get()).write(self.logger)
+
+    def load_configuration(self):
+        """
+        Load the persisted configuration, upgrading the configuration format
+        if an older version is detected.
+        """
+        self._config_path = self._path.child(b"current_configuration.json")
+        v1_config_path = self._path.child(
```

Comment:
:question: we don't want to keep the old `.v1.json` naming convention though? but we do need to see if we're dealing with an old v1 config that needs to be upgraded and such a config will have the old filename.